clients/drcachesim/analyzer.cpp

```diff
@@ -510,11 +528,10 @@ analyzer_tmpl_t<RecordType, ReaderType>::process_tasks(analyzer_worker_data_t *w
         if (sched_by_time_)
             cur_micros = get_current_microseconds();
         if (status == sched_type_t::STATUS_WAIT) {
-            // TODO i#5694: We'd like the forthcoming schedule_stats tool to know about
-            // waits and idle periods (to record "-" in its string): should the analyzer
-            // insert a new marker type that doesn't count toward ordinals (or else it
-            // needs a scheduler API to inject it)?
-            continue;
+            // We let tools know about waits so they can analyze the schedule.
+            // We synthesize a record here.  If we wanted this to count toward output
+            // stream ordinals we would need to add a scheduler API to inject it.
+            set_wait_record(record);
```

Comment:
What does a wait record on its own signify? There are no units or other ordering to denote how long the wait is/was.

Reply:
One wait per query to the scheduler.  I updated the docs to say "  A new marker is emitted each
     * time the tool analysis framework requests a new record from the scheduler and
     * is given a wait status."